client/homebrew/pages/sharePage/sharePage.jsx

```diff
@@ -1,137 +1,131 @@
 require('./sharePage.less');
 const React = require('react');
-const createClass = require('create-react-class');
+const { useState, useEffect, useCallback } = React;
 const { Meta } = require('vitreum/headtags');
 
 const Nav = require('naturalcrit/nav/nav.jsx');
 const Navbar = require('../../navbar/navbar.jsx');
 const MetadataNav = require('../../navbar/metadata.navitem.jsx');
 const PrintNavItem = require('../../navbar/print.navitem.jsx');
 const RecentNavItem = require('../../navbar/recent.navitem.jsx').both;
-const VaultNavItem = require('../../navbar/vault.navitem.jsx');
 const Account = require('../../navbar/account.navitem.jsx');
 const BrewRenderer = require('../../brewRenderer/brewRenderer.jsx');
 
 const { DEFAULT_BREW_LOAD } = require('../../../../server/brewDefaults.js');
 const { printCurrentBrew, fetchThemeBundle } = require('../../../../shared/helpers.js');
 
-const SharePage = createClass({
-	displayName     : 'SharePage',
-	getDefaultProps : function() {
-		return {
-			brew        : DEFAULT_BREW_LOAD,
-			disableMeta : false
-		};
-	},
-
-	getInitialState : function() {
-		return {
-			themeBundle                : {},
-			currentBrewRendererPageNum : 1
-		};
-	},
-
-	componentDidMount : function() {
-		document.addEventListener('keydown', this.handleControlKeys);
+const SharePage = (props)=>{
+	const { brew = DEFAULT_BREW_LOAD, disableMeta = false } = props;
 
-		fetchThemeBundle(this, this.props.brew.renderer, this.props.brew.theme);
-	},
+	const [state, setState] = useState({
+		themeBundle                : {},
+		currentBrewRendererPageNum : 1,
+	});
 
-	componentWillUnmount : function() {
-		document.removeEventListener('keydown', this.handleControlKeys);
-	},
+	const handleBrewRendererPageChange = useCallback((pageNumber)=>{
+		updateState({ currentBrewRendererPageNum: pageNumber });
+	}, []);
 
-	handleBrewRendererPageChange : function(pageNumber){
-		this.setState({ currentBrewRendererPageNum: pageNumber });
-	},
-
-	handleControlKeys : function(e){
+	const handleControlKeys = useCallback((e)=>{
 		if(!(e.ctrlKey || e.metaKey)) return;
 		const P_KEY = 80;
-		if(e.keyCode == P_KEY){
-			if(e.keyCode == P_KEY) printCurrentBrew();
+		if(e.keyCode === P_KEY) {
+			printCurrentBrew();
 			e.stopPropagation();
 			e.preventDefault();
 		}
-	},
+	}, []);
+
+	useEffect(()=>{
+		document.addEventListener('keydown', handleControlKeys);
```

Comment:
This will add a duplicate event listener every time brew.renderer, brew.theme, or handleControlkeys changes.